src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,112 +183,570 @@ def __repr__(self):
+
+        featurevariants.append(variant_dict)
+    return featurevariants
+
+
 def map_featureflag_to_keyvalue(featureflag):
-    '''
-        Helper Function to convert FeatureFlag object to KeyValue object
+    """
+    Helper Function to convert FeatureFlag object to KeyValue object
 
-        Args:
-            featureflag - FeatureFlag object to be converted
+    Args:
+        featureflag - FeatureFlag object to be converted
 
-        Return:
-            KeyValue object
-    '''
+    Return:
+        KeyValue object
+    """
     try:
         enabled = False
         if featureflag.state in ("on", "conditional"):
             enabled = True
 
-        feature_flag_value = FeatureFlagValue(id_=featureflag.name,
-                                              description=featureflag.description,
-                                              enabled=enabled,
-                                              conditions=featureflag.conditions)
-
-        set_kv = KeyValue(key=featureflag.key,
-                          label=featureflag.label,
-                          value=json.dumps(feature_flag_value,
-                                           default=lambda o: o.__dict__,
-                                           ensure_ascii=False),
-                          content_type=FeatureFlagConstants.FEATURE_FLAG_CONTENT_TYPE,
-                          tags={})
+        allocation = None
+        variants = None
+        telemetry = None
+        display_name = None
+
+        if hasattr(featureflag, FeatureFlagConstants.ALLOCATION):
+            allocation = featureflag.allocation
+
+        if hasattr(featureflag, FeatureFlagConstants.VARIANTS):
+            variants = featureflag.variants
+
+        if hasattr(featureflag, FeatureFlagConstants.TELEMETRY):
+            telemetry = featureflag.telemetry
+
+        if hasattr(featureflag, FeatureFlagConstants.DISPLAY_NAME):
+            display_name = featureflag.display_name
+
+        feature_flag_value = FeatureFlagValue(
+            id_=featureflag.name,
+            description=featureflag.description,
+            enabled=enabled,
+            conditions=featureflag.conditions,
+            display_name=display_name,
+            allocation=allocation,
+            variants=variants,
+            telemetry=telemetry,
+        )
+
+        set_kv = KeyValue(
+            key=featureflag.key,
+            label=featureflag.label,
+            value=json.dumps(
+                feature_flag_value, default=lambda o: o.__dict__, ensure_ascii=False
+            ),
+            content_type=FeatureFlagConstants.FEATURE_FLAG_CONTENT_TYPE,
+            tags={},
+        )
 
         set_kv.locked = featureflag.locked
         set_kv.last_modified = featureflag.last_modified
 
     except ValueError as exception:
-        error_msg = "Exception while converting feature flag to key value: {0}\n{1}".format(featureflag.key, exception)
+        error_msg = (
```

Comment:
nit: extra brackets

Reply:
Removed. I formatted the code using Vs code it seems it added the extra brackets. Will remove all extra brackets.